src/main/java/org/helpberkeley/memberdata/Main.java

```diff
@@ -709,27 +709,35 @@ private static void doUpdateMemberData(
         ApiQueryResult apiQueryResult = HBParser.parseQueryResult(json);
         Map<String, DetailsPost> deliveryDetails = HBParser.deliveryDetails(apiQueryResult);
 
-        String workflowFileName = request.uploadFile.getFileName();
-        String deliveries = apiClient.downloadFile(workflowFileName);
+        String originalWorkflowFileName = request.uploadFile.getOriginalFileName();
+        String updatedFileName = originalWorkflowFileName.replace(".csv", "-upd.csv");
+        String fileName = request.uploadFile.getFileName();
+        String deliveries = apiClient.downloadFile(fileName);
         request.postStatus(WorkRequestHandler.RequestStatus.Processing, "");
 
-        WorkflowParser parser = WorkflowParser.create(Collections.emptyMap(), deliveries);
-        WorkflowExporter exporter = new WorkflowExporter(parser);
+        WorkflowParser parser;
+        WorkflowExporter exporter;
         String updatedCSVData;
         try {
+            parser = WorkflowParser.create(Collections.emptyMap(), deliveries);
+            exporter = new WorkflowExporter(parser);
```

Comment:
Create the parser as an argument to the WorkflowExporter.create() call, as shown in my previous comment.  There is no need for a parser variable.  It isn't referenced at this level.